### ISSUE TITLE
(iOS) SyntaxError in plugin JS console messages

### ISSUE THREAD

**Issue by webcodepl** [dpa99c/cordova-plugin-firebasex#401]
My configuration:

Model: iPhone X
Version: iOS 13.4.1

Model: MacBook Pro 13
Version: 10.15.4

Cordova 9.0.0 (cordova-lib@9.0.1)

cordova-custom-config 5.1.0 "cordova-custom-config"
cordova-plugin-androidx 1.0.2 "cordova-plugin-androidx"
cordova-plugin-androidx-adapter 1.1.1 "cordova-plugin-androidx-adapter"
cordova-plugin-customfcmreceiver 1.0.0 "Custom FCM Receiver"
cordova-plugin-dialogs 1.3.2-dev "Notification"
cordova-plugin-enable-multidex 0.2.0 "Enable Multidex"
cordova-plugin-firebasex 9.1.1 "Google Firebase Plugin"
cordova-plugin-whitelist 1.3.4 "Whitelist"

Xcode Version 11.4.1 

Problem:

After receiving message i get error on console: SyntaxError: Unexpected EOF. I am sending message using:
npm run-script send -- --message=ios_data_notification.json --token="..."

![ios fcm bug](https://user-images.githubusercontent.com/11521576/81912686-35b45100-95cf-11ea-9d8c-9afa15191188.jpg)

I was trying also versions 8.0.1, 9.1.1-cli, compiling from xcode and xcodebuild, still the same error.

**Comment by dpa99c** [comment]
I can repro this issue:
It's actually an issue with escaping of line endings for Javascript when sending log messages from the native iOS implementation to the Cordova Webview for display in the JS console.

For example, an expected log message would be:

```javascript
console.log("FirebasePlugin[native]: didReceiveRemoteNotification: {\
    aps =     {\
        \"content-available\" = 1;\
    };\
    \"gcm.message_id\" = 1589924900140911;\
    \"key_1\" = \"Data for key one\";\
    messageType = data;\
    \"notification_body\" = \"Data message body\";\
    \"notification_foreground\" = true;\
    \"notification_ios_badge\" = 1;\
    \"notification_ios_sound\" = \"blackberry.caf\";\
    \"notification_title\" = \"Data message title\";\
}")
```

but the actual string being generated is:

```javascript
console.log("FirebasePlugin[native]: didReceiveRemoteNotification: {
    aps =     {
        \"content-available\" = 1;
    };
    \"gcm.message_id\" = 1589924900140911;
    \"key_1\" = \"Data for key one\";
    messageType = data;
    \"notification_body\" = \"Data message body\";
    \"notification_foreground\" = true;
    \"notification_ios_badge\" = 1;
    \"notification_ios_sound\" = \"blackberry.caf\";
    \"notification_title\" = \"Data message title\";
}")
```

i.e. no escaping backslashses at the end of lines, resulting in the console error `SyntaxError: Unexpected EOF`.

**Comment by echonox** [comment]
@dpa99c it seems you accidentally reverted this change in: 116f5ce36a15a3d46f543e53b6c6ee3d80721489

could you plz merge it again?

**Comment by echonox** [comment]
@dpa99c I tried your fix but it doesn't seem to work fully I get: `SyntaxError: Invalid escape in identifier: 'summary\'` (my payload data contains object with key summary)

**Comment by echonox** [comment]
@dpa99c  the fix doesn't work I am getting: `SyntaxError: Invalid escape in identifier: 'summary\'` my push data contains an object which has the key summary

**Comment by dpa99c** [comment]
I will look at it again when I have some time

**Comment by dpa99c** [comment]
@echonox It looks like your keys are wrapped in single quotes:

> SyntaxError: Invalid escape in identifier: 'summary\'

which is invalid JSON. 

Hence the Firebase Admin SDK will not allow a message to be sent containing single-quoted keys so I'm unable to reproduce your issue.

Can you confirm whether your keys are double-quoted or single-quoted?
In fact, if possible, can you supply a sample notification message which is causing you this issue.

**Comment by dpa99c** [comment]
I believe this is resolved by #430
